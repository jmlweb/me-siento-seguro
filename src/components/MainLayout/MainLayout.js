import React from 'react';
import PropTypes from 'prop-types';

import styles from './MainLayout.module.css';

const MainLayout = ({ header, main, footer }) => (
  <div className={styles.wrapper}>
    <header className={styles.header}>{header}</header>
    <main className={styles.main}>{main}</main>
    <footer className={styles.footer}>{footer}</footer>
  </div>
);

MainLayout.propTypes = {
  header: PropTypes.node.isRequired,
  main: PropTypes.node.isRequired,
  footer: PropTypes.node.isRequired,
};

export default MainLayout;
