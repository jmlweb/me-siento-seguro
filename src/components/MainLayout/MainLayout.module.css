.wrapper {
  display: flex;
  flex-direction: column;
  width: 100%;
  min-height: 100vh;
}

@keyframes headerBg {
  0% {
    background: linear-gradient(270deg, #022b3a, #1f7a8c);
    background-size: 150%;
  }
  12% {
    background: linear-gradient(225deg, #022b3a, #1f7a8c);
    background-size: 145%;
  }
  25% {
    background: linear-gradient(180deg, #022b3a, #1f7a8c);
    background-size: 140%;
  }
  33% {
    background: linear-gradient(135deg, #022b3a, #1f7a8c);
    background-size: 135%;
  }
  50% {
    background: linear-gradient(90deg, #022b3a, #1f7a8c);
    background-size: 130%;
  }
  66% {
    background: linear-gradient(135deg, #022b3a, #1f7a8c);
    background-size: 135%;
  }
  75% {
    background: linear-gradient(180deg, #022b3a, #1f7a8c);
    background-size: 140%;
  }
  88% {
    background: linear-gradient(225deg, #022b3a, #1f7a8c);
    background-size: 145%;
  }
  100% {
    background: linear-gradient(270deg, #022b3a, #1f7a8c);
    background-size: 150%;
  }
}

.header {
  color: #fff;
  padding: 15px 20px;
  transition: all 0.3s ease-in;
  animation-name: headerBg;
  animation-duration: 20s;
  animation-iteration-count: infinite;
  animation-timing-function: cubic-bezier(0.075, 0.82, 0.165, 1);
  border-bottom: 2px solid #fff;
  box-shadow: 0 2px 14px rgba(0, 0, 0, 0.2);
  position: relative;
}

.main {
  flex: 1;
  background: #e1e5f2;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 100%;
}

.main > * {
  width: 100%;
}

.footer {
  color: #fff;
  background: #022b3a;
  text-align: center;
  padding: 20px;
  border-top: 2px solid #fff;
  box-shadow: 0 -2px 14px rgba(0, 0, 0, 0.2);
  position: relative;
}
