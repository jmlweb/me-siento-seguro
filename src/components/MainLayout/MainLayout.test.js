import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import MainLayout from './MainLayout';

describe('MainLayout', () => {
  const header = <div>Header</div>;
  const main = <div>Main</div>;
  const footer = <div>Footer</div>;
  const element = <MainLayout header={header} main={main} footer={footer} />;
  test('component renders', () => {
    const wrapper = shallow(element);
    expect(wrapper).toBeTruthy();
  });
  test('snapshot', () => {
    const tree = renderer.create(element).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
