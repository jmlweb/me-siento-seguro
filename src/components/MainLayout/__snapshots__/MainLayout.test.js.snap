// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainLayout snapshot 1`] = `
<div
  className="wrapper"
>
  <header
    className="header"
  >
    <div>
      Header
    </div>
  </header>
  <main
    className="main"
  >
    <div>
      Main
    </div>
  </main>
  <footer
    className="footer"
  >
    <div>
      Footer
    </div>
  </footer>
</div>
`;
